src/nacl/bindings/crypto_pwhash_scryptsalsa208sha256.py

```diff
@@ -0,0 +1,142 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import, division, print_function
+
+from nacl._sodium import ffi, lib
+from nacl.exceptions import CryptoError
+
+
+crypto_pwhash_scryptsalsa208sha256_SALTBYTES = \
+    lib.crypto_pwhash_scryptsalsa208sha256_saltbytes()
+crypto_pwhash_scryptsalsa208sha256_STRBYTES = \
+    lib.crypto_pwhash_scryptsalsa208sha256_strbytes()
+crypto_pwhash_scryptsalsa208sha256_OPSLIMIT_INTERACTIVE = \
+    lib.crypto_pwhash_scryptsalsa208sha256_opslimit_interactive()
+crypto_pwhash_scryptsalsa208sha256_MEMLIMIT_INTERACTIVE = \
+    lib.crypto_pwhash_scryptsalsa208sha256_memlimit_interactive()
+crypto_pwhash_scryptsalsa208sha256_OPSLIMIT_SENSITIVE = \
+    lib.crypto_pwhash_scryptsalsa208sha256_opslimit_sensitive()
+crypto_pwhash_scryptsalsa208sha256_MEMLIMIT_SENSITIVE = \
+    lib.crypto_pwhash_scryptsalsa208sha256_memlimit_sensitive()
+
+OPSLIMIT_INTERACTIVE = \
+    crypto_pwhash_scryptsalsa208sha256_OPSLIMIT_INTERACTIVE
+MEMLIMIT_INTERACTIVE = \
+    crypto_pwhash_scryptsalsa208sha256_MEMLIMIT_INTERACTIVE
+OPSLIMIT_SENSITIVE = \
+    crypto_pwhash_scryptsalsa208sha256_OPSLIMIT_SENSITIVE
+MEMLIMIT_SENSITIVE = \
+    crypto_pwhash_scryptsalsa208sha256_MEMLIMIT_SENSITIVE
+SALTBYTES = \
+    crypto_pwhash_scryptsalsa208sha256_SALTBYTES
+STRBYTES = \
+    crypto_pwhash_scryptsalsa208sha256_STRBYTES
+
+
+def crypto_pwhash_scryptsalsa208sha256(outlen, passwd, salt,
+                                       opslimit=OPSLIMIT_SENSITIVE,
+                                       memlimit=MEMLIMIT_SENSITIVE
+                                       ):
+    """
+    Derive a cryptographic key using the ``passwd`` and ``salt``
+    given as input.
+
+    The work factor can be tuned by by picking different
+    ``opslimit`` and ``memlimit``.
+
+    The constants
+        - :py:const:`.OPSLIMIT_INTERACTIVE`
+        - :py:const:`.MEMLIMIT_INTERACTIVE`
+        - :py:const:`.OPSLIMIT_SENSITIVE`
+        - :py:const:`.MEMLIMIT_SENSITIVE`
+
+    are provided as a guidance for correct settings respectively for the
+    interactive login and the long term key protecting sensitive data
+    usage cases.
+
+    :param int outlen: int
+    :param bytes passwd: bytes
+    :param bytes salt: *must* be *exactly* :py:const:`.SALTBYTES` long
+    :param int opslimit:
+    :param int memlimit:
+    :rtype: bytes
+    """
+
+    if len(salt) != SALTBYTES:
+        raise ValueError("Invalid salt")
+
+    buf = ffi.new("unsigned char[]", outlen)
+
+    ret = lib.crypto_pwhash_scryptsalsa208sha256(buf, outlen, passwd,
+                                                 len(passwd), salt,
+                                                 opslimit, memlimit)
+
+    if ret != 0:
+        raise CryptoError("Key derivation fails!")
```

Comment:
Hopefully unreachable, since scryptsalsa should never return a non-zero result. #223 could be useful both here and at line 119